components/TextFields/src/MDCTextInputControllerBase.m

```diff
@@ -397,13 +397,16 @@ - (void)updateLeadingUnderlineLabel {
 #pragma mark - TextInput Customization
 
 - (void)updateTextInput {
-  UIFont *font = self.textInputFont;
-  if (self.mdc_adjustsFontForContentSizeCategory) {
-    font = [font mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleBody1
-                              scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
-    // TODO: (#4331) This needs to be converted to the new text scheme.
-  }
-  self.textInput.font = font;
+	UIFont *font = self.textInputFont;
+	if (self.mdc_adjustsFontForContentSizeCategory) {
+		font = [font mdc_fontSizedForMaterialTextStyle:MDCFontTextStyleBody1
+								  scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
+		// TODO: (#4331) This needs to be converted to the new text scheme.
+	}
+	
+	if (![self.textInput.font mdc_isSimplyEqual:font]){
+		self.textInput.font = font;
```

Comment:
From this change, it looks like there is a bug when setting the MDCTextField's `font` property. Could you please describe in the PR summary why the fix needs to be made here and not where the side-effect takes place?

Reply:
Awesome, thanks for confirming!